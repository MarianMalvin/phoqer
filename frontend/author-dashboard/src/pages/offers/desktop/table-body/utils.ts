import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
