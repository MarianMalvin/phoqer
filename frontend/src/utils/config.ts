import { IConfig, Themes } from '../interfaces';

const ws = process.browser
    ? `ws://${window.location.host.includes('localhost') ? 'dev.phoqer.com' : window.location.host}/api/`
    : 'ws://phoqer.com/api/';

const config = {
    socketUrl: (version = 'v1'): string => ws + version,
    uploadsUrl: (version = 'v2'): string => `/${version}/upload`,
    host: (lang = 'pl'): string => `/${lang}`,
    img: 'http://phoqer.com',
    themes: ['white', 'black'] as Themes[],
    googleApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAP_API_KEY,
    appSecretKey: process.env.NEXT_PUBLIC_SECRET_KEY,
    siteConfig: {
        warning: true,
        hideSearchFilters: false,
        hideTopSearchQuery: false,
        hideTopOffers: true,
        offerCardSize: 'big',
    } as IConfig,
};

export default config;
