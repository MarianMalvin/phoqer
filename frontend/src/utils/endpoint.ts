const endpointsMap = {
    ssr: (path = ''): string => (process.env.NEXT_PUBLIC_API_SSR || 'http://dev.phoqer.com/api/v2') + path,
    browser: (path = ''): string => (process.env.NEXT_PUBLIC_API_BROWSER || '') + path,
};

const endpoint = endpointsMap[process.browser ? 'browser' : 'ssr'];

export default endpoint;
