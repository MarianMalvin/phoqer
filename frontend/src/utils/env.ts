const env = {
    NEXT_PUBLIC_URL: process.env.NEXT_PUBLIC_URL || 'http://dev.phoqer.com',
    NEXT_PUBLIC_WS: process.env.NEXT_PUBLIC_WS || 'ws://dev.phoqer.com/api/v1',
    NEXT_PUBLIC_GOOGLE_MAP_API_KEY: process.env.NEXT_PUBLIC_GOOGLE_MAP_API_KEY || '',
    NEXT_PUBLIC_SECRET_KEY: process.env.NEXT_PUBLIC_SECRET_KEY || '',
};

export default env;
