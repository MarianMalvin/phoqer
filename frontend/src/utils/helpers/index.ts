export { mailRegex, passwordRegex, numberValidation, isEmpty } from './validation';
export { addZeroToNumber, moneyFormat, declOfNum } from './formatting';
export { dateFromTimestamp, addMonthToDate, formatTimestamp } from './date';
export { cutString } from './string';
export { throttle } from './lodash';
export { parseCookie, serverCookie, serverRedirect } from './ssr';
export { logger } from './logger';
export { formatCatList, findCategory, findSubCategory, findParentCategory } from './categories';
export { onlineStatus, timeIndicator } from './online';
