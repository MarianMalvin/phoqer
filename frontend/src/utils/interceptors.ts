import axios from 'axios';
import * as AxiosLogger from 'axios-logger';

const api = axios.create({});

api.interceptors.request.use(
    req => AxiosLogger.requestLogger(req, { data: false }),
    error => Promise.reject(error),
);

api.interceptors.response.use(
    res => AxiosLogger.responseLogger(res, { data: false }),
    error => Promise.reject(error),
);

export default api;
