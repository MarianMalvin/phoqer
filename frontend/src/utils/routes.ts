import { ChatTypeEnum } from '../interfaces';

type ID = string | number | null;
type Primitive = string | number;

const routes = {
    root: '/',
    auth: {
        login: '/auth/login',
        join: '/auth/join',
        forgot_pass: '/auth/forgot-pass',
        confirmation: '/auth/confirmation',
    },
    profile: {
        public: (id: ID = ':profileId'): string => `/profile/${id}`,
        private: '/profile/private',
    },
    my_offers: (status = 'all'): string => `/my-offers/${status}`,

    chat: {
        list: (type: ChatTypeEnum = ChatTypeEnum.CLIENT): string => `/chat/${type}`,
        item: (chat: Primitive, type: ChatTypeEnum = ChatTypeEnum.CLIENT): string => `/chat/${type}/${chat}`,
        new: (offerId: Primitive): string => `/chat/new/${offerId}`,
    },
    settings: (status: Primitive = 'general'): string => `/settings/${status}`,
    notifications: '/notifications',
    referral: '/referral',
    analytics: '/analytics',
    offers: {
        single: (id: ID = ':offerId'): string => `/offers/${id}`,
        edit: (id: ID = ':offerId'): string => `/offers/edit/${id}`,
        new: (step: ID = ':step', offerId?: string): string => `/offers/new/${step}${offerId ? '?offerId=' + offerId : ''}`,
        list: '/offers',
    },
    static: {
        help: '/static/help',
        business: '/static/business',
        rules: '/static/rules',
        politic: '/static/politic',
        faq: '/static/faq',
        safety: '/static/safety',
        map: '/static/map',
    },
    favorite: '/favorite',
};

export default routes;
