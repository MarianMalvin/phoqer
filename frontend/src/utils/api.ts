import axios, { AxiosResponse } from 'axios';

import {
    ChatType,
    IAuth,
    ICategories,
    IChats,
    IComment,
    IMessages,
    IOfferCard,
    IOfferDynamic,
    IOfferPagination,
    IPagination,
    IPublicProfile,
    ISignup,
} from '../interfaces';
import { INewChat } from '../redux/chat/chats/interfaces';
import { IBody } from '../redux/offers/new_offer/interfaces';

import config from './config';

const url2 = config.baseUrl[process.env.NODE_ENV]('v2');

interface ICommentBody {
    body: string;
    replies_id?: number;
    offer_id?: string;
    images: string[];
}

const api = {
    // V2
    categories: {
        get: (): Promise<AxiosResponse<ICategories>> => axios.get(`${url2}/categories`),
    },
    auth: {
        login: (form: FormData): Promise<AxiosResponse<IAuth>> => axios.post(`${url2}/auth/login`, form),
        signup: (body: ISignup): Promise<AxiosResponse<void>> => axios.post(`${url2}/users/signup`, body),
    },
    comments: {
        list: (id: string): Promise<AxiosResponse<IComment[]>> => axios.get(`${url2}/comments/${id}`),
        create: (body: ICommentBody): Promise<AxiosResponse<void>> => axios.post(`${url2}/comments`, body),
        delete: (id: number): Promise<AxiosResponse<void>> => axios.delete(`${url2}/comments/${id}`),
        like: (id: number): Promise<AxiosResponse<void>> => axios.patch(`${url2}/comments/${id}/like`),
        dislike: (id: number): Promise<AxiosResponse<void>> => axios.patch(`${url2}/comments/${id}/dislike`),
    },
    offers: {
        myOffers: ({ tab, params }: { tab: string; params: { [key: string]: unknown } }): Promise<AxiosResponse<IOfferDynamic>> =>
            axios.get(`${url2}/offers/status/${tab}`, { params }),
        publicOffers: (id: string, params: { [key: string]: unknown }): Promise<AxiosResponse<IOfferPagination>> =>
            axios.get(`${url2}/offers/public/${id}`, { params }),
        popular: (): Promise<AxiosResponse<IOfferPagination>> => axios.get(`${url2}/offers/popular`),
        single: (id: string): Promise<AxiosResponse<IOfferCard>> => axios.get(`${url2}/offers/${id}`),
        new: (body: IBody): Promise<AxiosResponse<void>> => axios.post(`${url2}/offers`, body),
        update: (id: string, body: IBody | Partial<Body>): Promise<AxiosResponse<void>> =>
            axios.patch(`${url2}/offers/${id}`, body),
        status: (id: string, body: { [key: string]: unknown }): Promise<AxiosResponse<void>> =>
            axios.patch(`${url2}/offers/status/${id}`, body),
        search: (params: { [key: string]: unknown }): Promise<AxiosResponse<IOfferPagination>> =>
            axios.get(`${url2}/offers/search`, { params }),
        deleteOffer: (id: string): Promise<AxiosResponse<void>> => axios.delete(`${url2}/offers/${id}`),
        favorite: {
            get: (): Promise<AxiosResponse<IOfferCard[]>> => axios.get(`${url2}/favorite`),
            patch: (id: string): Promise<AxiosResponse<IOfferCard[]>> => axios.patch(`${url2}/favorite/${id}`),
        },
    },
    profiles: {
        private: {
            user: (): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/me`),
            userUpdate: (body: Partial<IPublicProfile>): Promise<AxiosResponse<IPublicProfile>> =>
                axios.patch(`${url2}/users/me`, body),
        },
        public: {
            get: (id: number): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/${id}`),
            userShort: (id: number): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/short/${id}`),
        },
    },
    chat: {
        chats: (page = 1, type: ChatType = 'i_am_client'): Promise<AxiosResponse<IPagination<IChats>>> =>
            axios.get(`${url2}/chats?${type}=true`, { params: { page } }),
        createChat: (body: INewChat): Promise<AxiosResponse<INewChat>> => axios.post(`${url2}/chats`, body),
        messages: (id: number, page = 1): Promise<AxiosResponse<IPagination<IMessages>>> =>
            axios.get(`${url2}/chats/${id}`, { params: { page } }),
        offerInfo: (id: number): Promise<AxiosResponse<IOfferCard>> => axios.get(`${url2}/offers/offers/${id}`),
    },
    locations: {
        countries: (): Promise<AxiosResponse<void>> => axios.get(`${url2}/locations/countries`),
        cities: (slug: string): Promise<AxiosResponse<void>> => axios.get(`${url2}/locations/cities/${slug}`),
    },
    uploads: (form: FormData): Promise<AxiosResponse<{ image_url: string }>> =>
        axios.post(`${url2}/upload`, form, { headers: { 'content-type': 'multipart/form-data' } }),
};

export default api;
