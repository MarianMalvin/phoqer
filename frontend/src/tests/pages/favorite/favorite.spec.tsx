import Favorite from '../../../pages/favorite';
import { render } from '../../tests.utils';

describe('Favorite component', () => {
    it('match snapshot', () => {
        const { asFragment } = render(<Favorite />);
        expect(asFragment()).toMatchSnapshot();
    });
});
