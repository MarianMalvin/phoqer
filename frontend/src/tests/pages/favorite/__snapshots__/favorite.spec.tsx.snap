// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite component match snapshot 1`] = `<DocumentFragment />`;
